modules/CardStack.js

```diff
@@ -21,7 +21,12 @@ const buildNavigationState = (
   entries: Array<Location>,
   cards: Array<Card>,
 ): NavigationState<{}> => {
-  return entries.reduce(
+  // Find last route and reproduce actual route stack
+  // Fix > https://github.com/LeoLeBras/react-router-navigation/issues/37
+  const lastRouteIndex = entries.findIndex((entry) => entry.pathname === location.pathname)
+  const nextEntries = entries.slice(0, lastRouteIndex + 1)
```

Comment:
It seems more appropriate to rename this var `initialEntries`

Reply:
Thanks for review.
Agree. I Will fix it.